pyomo/mpec/complementarity.py

```diff
@@ -357,13 +357,18 @@ def construct(self, data=None):
         """
         Construct the expression(s) for this complementarity condition.
         """
-        if is_debug_set(logger):
-            logger.debug("Constructing complementarity list %s", self.name)
         if self._constructed:
             return
-        timer = ConstructionTimer(self)
         self._constructed = True
 
+        timer = ConstructionTimer(self)
+        if is_debug_set(logger):
+            logger.debug("Constructing complementarity list %s", self.name)
+
+        if self._anonymous_sets is not None:
```

Comment:
After seeing these three lines of code _everywhere_, I wondering if it wouldn't make sense to have some sort of helper function instead to call? If this ever changes, you've got dozens of places to change it.

Reply:
I had the same comment and opened #3136 to track moving this to the base class.